File: Problem/Easy/Problem_3/Solution.cpp
// C++ program using to calculate the mean of array using recursion

#include <iostream>

#define MAX 100

// Recursive function
float calculateMeanArray(int a[MAX], int n)
{
    if (n == 1)
        return a[0];
    return float(calculateMeanArray(a, n - 1) * (n - 1) + a[n - 1]) / n;
}

// Main function
int main()
{
    int a[6] = { 1, 2, 3, 4, 5, 6 };
    int n = 6;
    std::cout << "Mean of the array is: " << calculateMeanArray(a, n);
    return 0;
}

File: Problem/Easy/Problem_3/Solution.py
# Python3 program using for calculate the mean of array using recursion

# Recursive function
def calculateMeanOfArray(arr : list, n : int) -> float:
    if n == 1:
        return float(arr[n-1])
    return float((calculateMeanOfArray(arr, n - 1) * (n - 1) + arr[n - 1]) / n)

# Main
if __name__ == '__main__':
    arr = [1, 2, 3, 4, 5, 6]
    n = len(arr)
    print('MEAN of the array is: {}'.format(calculateMeanOfArray(arr, n)))